src/core/mac/mac_frame.cpp

```diff
@@ -896,6 +912,143 @@ const uint8_t *Frame::GetFooter(void) const
     return GetPsdu() + GetPsduLength() - GetFooterLength();
 }
 
+void Frame::GenerateImmAck(const Frame *aFrame, bool aIsFramePending)
+{
+    memset(this->mPsdu, 0, this->mLength);
```

Comment:
My solution for it would be only clearing the frame control field. 
When generating the frame control of a new frame I directly **OR** some bits, which caused that some dirty bits(if not clear) might effect the new frame. But for other fields, the values are set as a whole and the original values would be overriden. So I think it's ok to only reset the frame control field when generating.

Reply:
I think the using suggestion below would be better:

> For preparing the frame, how about preparing the FCF in a local variable here and and then using Encoding::LittleEndian::WriteUint16() to write it in mPsdu

Then we won't need to clear anything.